src/base/base_runner.js

```diff
@@ -4,13 +4,55 @@ import shuffle from 'lodash.shuffle';
 import type {
   Logger,
   IRegistry,
+  Hooks,
 } from '../../types';
 
 class BaseRunner {
+  preProcess: () => Promise<void>;
+  healthCheck: () => Promise<void>;
+  terminationCheck: () => Promise<boolean>;
+  errorCount: number;
   registry: IRegistry;
   logger: Logger;
 
-  getActiveSubsciptions(topics: Array < string > = null): Array < string > {
+  constructor(hooks: Hooks = {}) {
+    this.errorCount = 0;
+    this.preProcess = hooks.preProcess || (() => Promise.resolve());
+    this.healthCheck = hooks.healthCheck || (() => Promise.resolve());
+    this.terminationCheck = hooks.terminationCheck || (() => Promise.resolve(false));
+    /* eslint-disable no-console */
+    this.logger = {
```

Comment:
do we need to keep the logging?

Reply:
@johndagostino just makes it a little smoother in regards to typing. no hard requirement. at the moment we're using base_runner as an abstract class, but maybe it should just be an interface or something?

Reply:
@bassjacob nullLogger is already in the package?